buildSrc/subprojects/packaging/src/main/kotlin/org/gradle/gradlebuild/packaging/ShadedJar.kt

```diff
@@ -31,31 +32,27 @@ import java.nio.file.Path
 import java.util.*
 import java.util.jar.JarFile
 import java.util.jar.JarOutputStream
+import javax.inject.Inject
 
 
-open class ShadedJar : DefaultTask() {
-    @InputFiles
-    val relocatedClassesConfiguration = project.files()
-    @InputFiles
-    val classTreesConfiguration = project.files()
-    @InputFiles
-    val entryPointsConfiguration = project.files()
-    @InputFiles
-    val manifests = project.files()
-
-    /**
-     * The build receipt properties file.
-     *
-     * The file will be included in the shaded jar under {@code /org/gradle/build-receipt.properties}.
-     */
-    @InputFile
-    val buildReceiptFile: RegularFileProperty = project.layout.fileProperty()
-
-    /**
-      * The output Jar file.
-      */
-    @OutputFile
-    val jarFile = newOutputFile()
```

Comment:
❌ Please do not replace the `jarFile` by a file property which is not created by `newOutputFile`. Doing so causes the link between the task and the output file property to be lost. I.e. when adding the Property as in input to another task, no task dependency is added.